client/src/js/services/CommonModuleState.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+/**
+ * @description
+ * Very simple service to define state variables that are common to most
+ * modules but are defined under different names and structures. This service is
+ * essentially a semantic wrapper around many common states.
+ *
+ * @returns {object} Getters and Setters for common modules states
+ *
+ * @todo Refactor API to remove redundant states
+ *
+ * @todo Using getters and setters to semantically expose variables may be more
+ * costly than it is worth, this should be investigated
+ */
+
+angular.module('bhima.services').factory('ModuleState', ModuleState);
+    
+ModuleState.$inject = ['$timeout'];
+
+function ModuleState ($timeout) { 
```

Comment:
I really like this idea of module states.  It seems that you've limited it to what seems to be more mechanical states (valid, invalid, success, loading).  Would you entertain something more flexible such as  state.setState('searching'), state.setState('orderReview'), etc?

It seems like lots of our modules may wish to use states that aren't directly related to data/controller validation, but are more tied to what the user is presented (a table of values versus a graph versus a form to input new data).  Is that beyond the scope of this Factory?